api/provision/async-provisioning-check.ts

```diff
@@ -74,15 +75,18 @@ async function makeRequest(
 async function baseHandler(event: SQSEvent): Promise<SQSBatchResponse> {
   // The item IDs for records that are still in the IN_PROGRESS or NOT_STARTED state
   const stillInProgress: string[] = [];
-  // Items that have moved to the COMPELTE or FAILED state
+  // Items that have moved to the COMPLETE or FAILED state
   const moveToReady: ProvisionCspResponse[] = [];
   if (!event?.Records?.length) {
     logger.info("There are no records in this request.");
   }
   for (const record of event.Records) {
     const request = JSON.parse(record.body) as ProvisionCspResponse;
-    const response = request.content.response as AsyncProvisionResponse;
-    const client = await makeClient(response.$metadata.request.targetCsp.name);
+    let cspName = "";
+    if (request.targetCsp) {
+      cspName = request.targetCsp.name;
+    }
+    const client = await makeClient(cspName);
```

Comment:
I was thinking being an `""` or `"BlankCsp"` would result in the same due to the underlying `getConfiguration()` being used in `makeClient()` and an error would be thrown (although need to probably uncomment that error at this point in time since we have real CSPs). The negative case can be handled explicitly and thrown here though.

Reply:
If passing in `""` is expected to return an error 100% of the time then we should probably catch it here.